### ISSUE TITLE
Custom URLs to Track Advertising Campaigns

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#901]
It would be nice to track the effectiveness of our advertising campaigns beyond what MailChimp and Google Analytics provides. For example, if we had the ability to put a campaign string into the url, then we could check to see how effective that particular campaign is. Say, something like, 'http://projectsidewalk.io/campaign=reddit_post_to_rdisability_on_2017_07_25' which, when loaded, would then quickly convert to simply 'http://projectsidewalk.io' so as to not have a confusing url at top.

**Comment by jonfroehlich** [comment]
@misaugstad are you going to work on this then? When?

**Comment by misaugstad** [comment]
@jonfroehlich Almost done :)

**Comment by manaswis** [comment]
Closed via PR #958 and deployed in v4.1.8 (PR #979)

**Comment by manaswis** [comment]
It can be tested via:
- [projectsidewalk.io/?referrer=test_campaign](http://projectsidewalk.io/?referrer=test_campaign)
- [projectsidewalk.io/?referrer=test_campaign&to=results](http://projectsidewalk.io/?referrer=test_campaign&to=results)

**Comment by maddalihanumateja** [comment]
I've tried to make custom urls will more flexible in the develop-add-mturker-role branch. The previous implementation had hardcoded the parameters that could be passed to the home endpoint. The 2 options were a referrer and a redirect url. However, when working with mturk we also need to pass other parameters such as hitId, workerId, and assignmentId. I've changed the query string to have a variable number of parameters in key value pair form. The application controller checks if specific parameters are present and then behaves accordingly. For example: when the referrer is mturk it checks for hitId etc . For other referrers it checks for a redirect url (redirects to home by default as in the previous implementation).
